src/main/java/io/vertx/core/datagram/impl/DatagramSocketImpl.java

```diff
@@ -210,16 +212,37 @@ public DatagramSocket resume() {
   @Override
   @SuppressWarnings("unchecked")
   public DatagramSocket send(Buffer packet, int port, String host, Handler<AsyncResult<DatagramSocket>> handler) {
+    Objects.requireNonNull(packet, "no null packet accepted");
     Objects.requireNonNull(host, "no null host accepted");
-    ChannelFuture future = channel().writeAndFlush(new DatagramPacket(packet.getByteBuf(), new InetSocketAddress(host, port)));
-    addListener(future, handler);
+    InetSocketAddress addr = InetSocketAddress.createUnresolved(host, port);
+    if (addr.isUnresolved()) {
```

Comment:
This will always be true. Creating an unresolved address will not even attempt to lookup in the local cache to see if it was previously resolved.